### PR TITLE
Additional filter texts

### PR DESCRIPTION
**Pull request by genlu**
Here's an example. The snippet is changed to has "cw" as regular `FilterText`, and "Console" and "WriteLine" as `AdditionalFilterTexts`

MultipleFilterTexts

### CODE REVIEW

File: src/Features/Core/Portable/Completion/CompletionItem.cs [hunk 1]
```diff
@@ -45,8 +47,21 @@ public sealed class CompletionItem : IComparable<CompletionItem>
         /// </summary>
         public string FilterText => _filterText ?? DisplayText;
 
+        internal ImmutableArray<string> AdditionalFilterTexts
+        {
+            get => _additionalFilterTexts;
+            set
+            {
+                Debug.Assert(!value.IsDefault && (value.IsEmpty || HasDifferentFilterText),
```
Comment (akhera99) [inline src/Features/Core/Portable/Completion/CompletionItem.cs:55]: I like this designation, makes sense to me.

**Review by akhera99 — COMMENTED** [review]
LGTM with tests!

**Review by akhera99 — APPROVED** [review]

**Comment by dibarbet** [comment]
@akhera99 @genlu we should think about bringing this to LSP as well, currently they only have a single filterText field
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItem

**Comment by genlu** [comment]
@dibarbet Sure, what's the process of making such a proposal?

**Comment by dibarbet** [comment]
@genlu we can file an issue on the LSP area path in ADO and ping Jose to bring to to an LSP sync meeting.  Once we've implemented it internally in the VS client and server we can consider porting to the public spec.

**Comment by genlu** [comment]
Sure will do. 

> Once we've implemented it internally in the VS client and server we can consider porting to the public spec.

The scenario that requires AdditionalFIiterTexts seems to be very limited at the moment, and everything that do require it is internal, so I'm a little concerned if this is something we want to publicly expose. Until at least snippets and AdditionalFIiterTexts  are made public

**Pull request merged**